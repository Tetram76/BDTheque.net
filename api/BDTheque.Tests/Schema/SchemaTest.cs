namespace BDTheque.Tests.Schema;

using BDTheque.Tests.Helpers;
using Snapshooter;

public class SchemaTest
{
    [Fact]
    public async Task SchemaChangeTest()
    {
        ISchema schema = await TestServices.Executor.GetSchemaAsync(default);
        schema.ToString().MatchSnapshot();
    }

    [Fact]
    public async Task GetGenresTest()
    {
        string response = await TestServices.ExecuteRequestAsync(
            b => b.SetQuery(
                """
                query {
                    genres(last: 10, where: {initiale: {eq: "A"}}) {
                        pageInfo {
                            hasNextPage
                            hasPreviousPage
                        }
                        nodes {
                            id
                            nom
                        }
                    }
                }
                """
            )
        );

        response.MatchSnapshot();
    }
}
