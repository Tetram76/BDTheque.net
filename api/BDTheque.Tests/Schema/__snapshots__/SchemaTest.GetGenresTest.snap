{
  "errors": [
    {
      "message": "Unexpected Execution Error",
      "locations": [
        {
          "line": 2,
          "column": 5
        }
      ],
      "path": [
        "genres"
      ],
      "extensions": {
        "message": "No service for type 'BDTheque.Data.Context.BDThequeContext' has been registered.",
        "stackTrace": "   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)\r\n   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)\r\n   at HotChocolate.Resolvers.Expressions.Parameters.ServiceHelper.<>c__DisplayClass7_3`1.<<UseResolverServiceInternal>b__5>d.MoveNext()\r\n--- End of stack trace from previous location ---\r\n   at HotChocolate.Resolvers.Expressions.Parameters.ServiceHelper.<>c__DisplayClass7_1`1.<<UseResolverServiceInternal>b__3>d.MoveNext()\r\n--- End of stack trace from previous location ---\r\n   at HotChocolate.Execution.Processing.Tasks.ResolverTask.ExecuteResolverPipelineAsync(CancellationToken cancellationToken)\r\n   at HotChocolate.Execution.Processing.Tasks.ResolverTask.TryExecuteAsync(CancellationToken cancellationToken)"
      }
    }
  ],
  "data": {
    "genres": null
  }
}
